src/Maker/MakeForgottenPassword.php

```diff
@@ -0,0 +1,320 @@
+        );
+        $input->setArgument(
+            'password-field',
+            $interactiveSecurityHelper->guessPasswordField($io, $userClass)
+        );
+    }
+
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+        // This recipe depends upon Doctrine ORM, to save the token and update the user
+        ORMDependencyBuilder::buildDependencies($dependencies);
+
+        $dependencies->addClassDependency(
+            AbstractType::class,
+            'form'
+        );
+        $dependencies->addClassDependency(
+            Validation::class,
+            'validator'
+        );
+        $dependencies->addClassDependency(
+            TwigBundle::class,
+            'twig-bundle'
+        );
+        $dependencies->addClassDependency(
+            SecurityBundle::class,
+            'security'
+        );
+        $dependencies->addClassDependency(
+            SwiftmailerBundle::class,
+            'mail'
+        );
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $userClass = $input->getArgument('user-class');
+        $userClassNameDetails = $generator->createClassNameDetails(
+            '\\'.$userClass,
+            'Entity\\'
+        );
+
+        // 1) Update the User class with a token we will use later
+        $classDetails = new ClassDetails($userClass);
+        $userManipulator = new ClassSourceManipulator(
+            file_get_contents($classDetails->getPath())
+        );
+        $userManipulator->setIo($io);
+        $userManipulator->addEntityField(
+            'resetPasswordToken',
+            [
+                'type' => 'string',
+                'nullable' => true,
+            ]
+        );
+        $userManipulator->addEntityField(
+            'passwordRequestedAt',
+            [
+                'type' => 'datetime',
+                'nullable' => true,
+            ]
+        );
+        $this->fileManager->dumpFile($classDetails->getPath(), $userManipulator->getSourceCode());
+
+        // 2) Generate the "username" form class
+        $usernameField = $input->getArgument('username-field');
+        $usernameFormClassDetails = $this->generateUsernameFormClass(
+            $userClassNameDetails,
+            $generator,
+            $usernameField
+        );
+
+        // 3) Generate the "new password" form class
+        $resettingFormClassDetails = $this->generateResettingFormClass(
+            $userClassNameDetails,
+            $generator
+        );
+
+        // 4) Generate the controller
+        $controllerClassNameDetails = $generator->createClassNameDetails(
+            'ForgottenPasswordController',
+            'Controller\\'
+        );
+
+        // We assume User entity has some password setter, and e-mail getter
+        // TODO: display a warning if not?
+        $passwordSetter = 'set'.ucfirst($input->getArgument('password-field'));
+        $emailGetter = 'getEmail';
```

Comment:
I *am* kinda thinking that we should make this work for "email address" only - as you suggested. So, instead of asking the user for their "username" field, we ask them for their `email` field, then use that everywhere. We should probably do some validation that the getter & setter exist for that field, and throw an error if it does not (if people don't like that, and also want the command to ask them for a custom getter & setter, we can handle that in the future IF people ask for it - not now).

Reply:
Ok, and it makes sense since reset password _requires_ an e-mail field. Modifying the request action to accept usernames too should be easy enough, I may add a comment in the generated code.